lib/starknet-jvm.md

```diff
@@ -220,37 +220,94 @@ fun main(args: Array<String>) {
 
 # Package com.swmansion.starknet.data.types
 
-Data classes representing StarkNet objects.
+The types module contains classes representing various data types used in StarkNet and utilities for
+interacting with them.
 
 # Package com.swmansion.starknet.data.types.transactions
 
-Data classes representing StarkNet transactions.
+The transactions submodule of the types module contains classes representing sent and received transactions
+as well as payloads of transactions to be sent. 
 
 # Package com.swmansion.starknet.provider
 
-Provider interface used for interacting with StarkNet.
+The provider module contains the Provider interface and its implementations. 
+The Provider interface defines methods for interacting with StarkNet,
+such as calling contracts and querying the state of the network.
+
+```java
+// Create a provider instance
+Provider provider = ...
+
+// Get a storage value request
+Request<Felt> request = provider.getStorageAt(address, key);
+// Send a request
+Felt response = request.send();
+
+// For most methods block hash, number or tag can be specified
+Request<Felt> request = provider.getStorageAt(address, key, Felt.fromHex("0x123..."));
+Request<Felt> request = provider.getStorageAt(address, key, 1234);
+Request<Felt> request = provider.getStorageAt(address, key, BlockTag.LATEST);
+```
 
 # Package com.swmansion.starknet.provider.exceptions
 
 Exceptions thrown by the StarkNet providers.
 
+`Request.send()` throws `RequestFailedException` which has to be 
+handled when writing in Java.
+
+```java
+Request<Felt> request = ...
+// Send a request
+try {
+    Felt response = request.send();
+} catch (RequestFailedException e) {
+    // Handle an exception
+}
+```
+
+In the case of `Request.sendAsync()`, an exception would have to be handled in the returned `CompletableFuture`.
+
 # Package com.swmansion.starknet.provider.gateway
 
 Provider utilising StarkNet gateway and feeder gateway for communication with the network.
 
+```java
+// Create a provider using a factory
+GatewayProvider.makeTestnetProvider();
+// Chain id can be specified
+GatewayProvider.makeTestnetProvider(StarknetChainId.TESTNET2);
+// As well as the custom HttpService
+GatewayProvider.makeTestnetProvider(myHttpService, StarknetChainId.TESTNET2);
+
+// Provider can be also created using a construcotr
+new GatewayProvider("feederGatewayUrl", "gatewayUrl", StarknetChainId.TESTNET);
+// or with a custom HttpService
+new GatewayProvider("feederGatewayUrl", "gatewayUrl", StarknetChainId.TESTNET, myHttpService); 
+```
+
 # Package com.swmansion.starknet.provider.rpc
 
 Provider implementing the [JSON RPC interface](https://github.com/starkware-libs/starknet-specs)
 to communicate with the network.
 
+```java
+// JsonRpcProvider can only be created using constructor
+new JsonRpcProvider("url", StarknetChainId.TESTNET);
+// or with a custom HttpService
+new JsonRpcProvider("url", StarknetChainId.TESTNET, myHttpService);
```

Comment:
Some users might think "url" is some special constant here. Can we name it "rpcNodeUrl"?

Reply:
I don't think having special constants like this is very popular in Java, but I guess we can change it.